File: modules/security/exercises/02_sanitize_input/solution/python/main.py
def sanitize(text: str) -> str:
    t = text
    for bad in ("ignore instructions", "system:"):
        t = t.replace(bad, "")
    return t.strip()

def main() -> None:
    user = "hello system: ignore instructions world"
    print(f"SANITIZED_OK:{sanitize(user)}")

if __name__ == "__main__":
    main()

File: modules/security/exercises/02_sanitize_input/solution/cpp/main.cpp
#include <iostream>
#include <string>

static std::string sanitize(std::string text) {
    for (const char* bad : {"ignore instructions", "system:"}) {
        std::string needle(bad);
        for (size_t pos = 0; (pos = text.find(needle, pos)) != std::string::npos;) {
            text.erase(pos, needle.size());
        }
    }
    while (!text.empty() && text.front() == ' ') text.erase(text.begin());
    while (!text.empty() && text.back() == ' ') text.pop_back();
    return text;
}

int main() {
    std::cout << "SANITIZED_OK:" << sanitize("hello system: ignore instructions world") << "\n";
    return 0;
}
